src/data/TotalSupply.ts

```diff
@@ -12,3 +17,42 @@ export function useTotalSupply(token?: Token): TokenAmount | undefined {
 
   return token && totalSupply ? TokenAmount.fromRawAmount(token, totalSupply.toString()) : undefined
 }
+
+export function useTotalSupplyV2(chainId: ChainId, tokenAddress: string): number {
+  const contract = useTokenContractForReading(tokenAddress, chainId)
+  const [totalSupply, setTotalSupply] = useState<number>(0)
+  const { readProvider } = useKyberSwapConfig(chainId)
+
+  const addressCheckSum = isAddress(chainId, tokenAddress)
+  const tokenContract = useContractForReading(addressCheckSum ? addressCheckSum : undefined, ERC20_ABI, chainId)
+
+  const fetchTotalSupply = useCallback(async () => {
+    const getTotalSupply = async (): Promise<number> => {
+      try {
+        if (!chainId || !readProvider) {
+          return 0
+        }
+
+        const rawTotalSupply = await contract?.totalSupply()
+        const decimals = await contract?.decimals()
+
+        return Number(
+          new Fraction(rawTotalSupply.toString(), JSBI.exponentiate(JSBI.BigInt(10), JSBI.BigInt(decimals))).toFixed(8),
+        )
+      } catch (e) {
+        return 0
+      }
+    }
+
+    const totalSupply = await getTotalSupply()
+    setTotalSupply(totalSupply)
+  }, [chainId, contract, readProvider])
+
+  useEffect(() => {
+    if (tokenContract) {
+      fetchTotalSupply()
+    }
```

Comment:
add cancel mechanism

Reply:
after searching, this is not used anywhere. removed